### PR TITLE
PP-2744 Create cards table

### PR DESCRIPTION
**Pull request by georgeracu**
- Added migration script;
- Added entity `CardEntity`;
- Added tests and fixtures;

with @alexbishop1

## WHAT
* Another step into splitting the `charges` table into smaller pieces. This table `cards` holds details regarding address, name and card details for a payment request. 
* It follows the pattern where we write in parallel in `charges` and `cards` but we read only from `charges`.
* It will be backfilled later on with data and we will switch to read from it as well.

### CODE REVIEW

File: src/main/java/uk/gov/pay/connector/model/domain/CardEntity.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+@Table(name = "cards")
+public class CardEntity extends AbstractEntity {
+    @Column(name = "last_digits_card_number")
+    private String lastDigitsCardNumber;
+
+    @Column(name = "cardholder_name")
+    private String cardHolderName;
+
+    @Column(name = "expiry_date")
+    private String expiryDate;
```
Comment (chrisgrimble) [inline src/main/java/uk/gov/pay/connector/model/domain/CardEntity.java:18]: Should be a ZonedDateTime and has a UTCDateTimeConverter on it.

File: src/main/resources/migrations.xml [hunk 2]
```diff
@@ -695,4 +695,54 @@
...
+            <column name="email" type="varchar(50)">
+                <constraints nullable="false"/>
+            </column>
+            <column name="last_digits_card_number" type="char(4)">
+                <constraints nullable="false"/>
+            </column>
+            <column name="cardholder_name" type="varchar(255)">
+                <constraints nullable="false"/>
+            </column>
+            <column name="expiry_date" type="varchar(6)">
```
Comment (chrisgrimble) [inline src/main/resources/migrations.xml:721]: Should be a timestamp without timezone not a varchar

**Review by chrisgrimble — CHANGES_REQUESTED** [review]
Please see comments in code about dates.

File: src/main/java/uk/gov/pay/connector/model/domain/CardEntity.java (hunk 1 above)
Comment (alexbishop1) [reply src/main/java/uk/gov/pay/connector/model/domain/CardEntity.java:18]: Expiry date is just a month and a year (it’s exactly like the “09/22” on the card). We store these as strings (we never treat them as dates, just send them to the payment gateway and show them in selfservice). So I think it’s fine as it is.

**Review by chrisgrimble — APPROVED** [review]
Comments explain why we would not want to save this as a date.

**Pull request merged**